lib/Runtime/Language/SourceTextModuleRecord.cpp

```diff
@@ -665,6 +705,23 @@ namespace Js
         return childModuleRecord;
     }
 
+    void SourceTextModuleRecord::InitializeLocalImports()
+    {
+        Assert(importRecordList != nullptr);
+
+        importRecordList->Map([&](ModuleImportOrExportEntry& importEntry) {
+            Js::PropertyId importName = EnsurePropertyIdForIdentifier(importEntry.importName);
+
+            SourceTextModuleRecord* childModule = this->GetChildModuleRecord(importEntry.moduleRequest->Psz());
+            ModuleNameRecord* importRecord = nullptr;
+            if (!childModule->ResolveExport(importName, nullptr, nullptr, &importRecord)
+                || importRecord == nullptr)
+            {
+                JavascriptError::ThrowSyntaxError(scriptContext, JSERR_ModuleResolveImport, importEntry.importName->Psz());
```

Comment:
probably not worth enter and throw here. you can just create errorobject and return and we'll just return the error object back.